src/nodes/NodeConnectionManager.ts

```diff
@@ -1677,6 +1728,9 @@ class NodeConnectionManager {
       connectionsEntry.authenticatedForward = AuthenticatingState.FAIL;
       this.authenticateFail(targetNodeIdString, err);
       return;
+    } finally {
+      await writer?.abort();
+      await reader?.cancel();
```

Comment:
Does this cause a problem if the writer and reader are already completed? I think cancellation should only be done in the case of an error.